intra_process_demo/src/image_pipeline/camera_node.hpp

```diff
@@ -90,6 +93,10 @@ class CameraNode : public rclcpp::Node
   std::thread thread_;
   std::atomic<bool> canceled_;
 
+  /// whether or not to add a watermark to the image showing process id and
+  /// pointer location
+  std::atomic<bool> watermark_;
```

Comment:
Are you swapping this during runtime? If not, then it probably doesn't need to be atomic, but if you are, then it is fine as-is.